string state_file
---
string result
